source/common/http/conn_manager_impl.cc

```diff
@@ -673,7 +673,7 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
 
   // Base headers.
   connection_manager_.config_.dateProvider().setDateHeader(headers);
-  headers.insertServer().value(connection_manager_.config_.serverName());
+  headers.insertServer().value().setStatic(connection_manager_.config_.serverName());
```

Comment:
Yeah it is. Static for lifetime of connection manager which must live for lifetime of connection.